lib/u2/ScrollDAOTable.es6.js

```diff
@@ -0,0 +1,208 @@
+    }
+
+    ^ col-head {
+      overflow: hidden;
+      white-space: nowrap;
+      text-overflow: ellipsis;
+      margin: auto 5px;
+      display: flex;
+      min-height: 0;
+      justify-content: space-between;
+    }
+
+    ^scroll-container {
+      position: relative;
+      flex-grow: 1;
+    }
+
+    ^ span, ^ .id {
+      overflow: hidden;
+      white-space: nowrap;
+      text-overflow: ellipsis;
+      margin: auto 5px;
+    }
+
+    ^ .id .overlay {
+      display: none;
+      position: absolute;
+      top: 0;
+      left: 0;
+    }
+
+    ^ li:hover .id {
+     color: transparent;
+    }
+
+    ^ li:hover .id .overlay {
+      height: 100%;
+      color: rgba(0, 0, 0, 0.8);
+      background-color: rgba(255, 255, 255, 0.6);
+      display: flex;
+      align-items: center;
+    }
+
+    ^icon {
+      z-index: 2;
+      display: inline-block;
+      padding: 1rem;
+    }
+
+    ^button {
+      flex-grow: 0;
+      flex-shrink: 0;
+      font-size: inherit;
+      background-color: inherit;
+      color: inherit;
+      padding: 0.5rem 0 0.5rem 0.5rem;
+      -webkit-user-select: none;
+      -moz-user-select: none;
+      -ms-user-select: none;
+      user-select: none;
+    }
+
+    ^button:hover {
+      background-color: rgba(255, 255, 255, 0.2);
+      cursor: pointer;
+    }
+  `,
+
+  classes: [
+    {
+      name: 'RowFormatter',
+      implements: ['foam.u2.RowFormatter'],
+
+      documentation: `Extend foam.u2.RowFormatter to support inversion of
+          control for rendering individual table cells.`,
+
+      imports: ['columns'],
+
+      methods: [
+        function format(data, opt_columns) {
+          const columns = opt_columns || this.columns;
+
+          let str = '';
+          for (let i = 0; i < columns.length; i++) {
+            const col = columns[i];
+            const value = data ? col.f(data) : undefined;
+            str += col.rawTableCellFormatter(data ? col.f(data) : undefined, data,
+                                          col);
+          }
+
+          return str;
+        },
+      ],
+    },
+  ],
+
+  properties: [
+    {
+      class: 'foam.u2.ViewSpec',
+      name: 'scrollView',
+      documentation: `View that actually performs scrolling. Default to
+          foam.u2.view.ScrollDAOView with two-way bound "data", reasonable row
+          defaults, and a per column/cell row formatter.`,
+      factory: function() {
+        return {
+          class: 'foam.u2.view.ScrollDAOView',
+          data$: this.data$,
+          numRows: 40,
+          rowHeight: 50,
+          rowFormatter: this.RowFormatter.create(),
+        };
+      },
+    },
+    {
+      class: 'Class',
+      name: 'of',
+      // TODO(markdittmer): This should be an expression, but that's not
+      // working.
+      factory: function() { return this.data.of; },
+    },
+  ],
+
+  methods: [
+    function initE() {
+      const self = this;
```

Comment:
nit: appears unused

Reply:
Used on lines 192, 193

Reply:
Ok. It's in a block that also uses this. Why have two names for the same thing?